src/libraries/System.Net.Http/src/System/Net/Http/StreamContent.cs

```diff
@@ -52,6 +53,19 @@ private void InitializeContent(Stream content, int bufferSize)
             if (NetEventSource.IsEnabled) NetEventSource.Associate(this, content);
         }
 
+        protected override void SerializeToStream(Stream stream, TransportContext? context, CancellationToken cancellationToken)
+        {
+            Debug.Assert(stream != null);
+
+            PrepareContent();
+
+            // Last chance to check for timeout/cancellation, sync Stream API doesn't have any support for it.
+            cancellationToken.ThrowIfCancellationRequested();
+
+            // If the stream can't be re-read, make sure that it gets disposed once it is consumed.
+            StreamToStreamCopy.Copy(_content, stream, _bufferSize, !_content.CanSeek);
```

Comment:
There has to be a way to cancel a slow response body read or the sync APIs will be fundamentally unusable. Any sync HTTP request could hang your app indefinitely and ignore your timeouts.

Reply:
As noted elsewhere, this is handled at a higher-level by tearing down the connection in response to a cancellation request.